build/source/engine/allocspace.f90

```diff
@@ -57,136 +56,194 @@ module allocspace_module
+ USE globalData,only: gru_struc                    ! gru-hru mapping structures
+ 
  implicit none
- ! input: model control
- integer(i4b),            intent(in)   :: nHRU          ! number of HRUs
- integer(i4b),            intent(in)   :: nSnow(:)      ! number of snow layers for each HRU
- integer(i4b),            intent(in)   :: nSoil(:)      ! number of soil layers for each HRU
- ! output: data structures
- type(var_i),             intent(out)  :: timeStruct    ! model time data
- type(spatial_double),    intent(out)  :: forcStruct    ! model forcing data
- type(spatial_double),    intent(out)  :: attrStruct    ! local attributes for each HRU
- type(spatial_int),       intent(out)  :: typeStruct    ! local classification of soil veg etc. for each HRU
- type(spatial_double),    intent(out)  :: mparStruct    ! model parameters
- type(spatial_intVec),    intent(out)  :: indxStruct    ! model indices
- type(spatial_doubleVec), intent(out)  :: progStruct    ! model prognostic (state) variables
- type(spatial_doubleVec), intent(out)  :: diagStruct    ! model diagnostic variables
- type(spatial_doubleVec), intent(out)  :: fluxStruct    ! model fluxes
- type(var_dlength),       intent(out)  :: derivStruct   ! model derivatives
- type(var_d),             intent(out)  :: bparStruct    ! basin-average parameters
- type(var_dlength),       intent(out)  :: bvarStruct    ! basin-average variables
- ! output: error control
- integer(i4b),intent(out)              :: err           ! error code
- character(*),intent(out)              :: message       ! error message
- ! ---------------------------------------------------------------------------------------------------------------
+ ! define output
+ integer(i4b),intent(out)             :: nGRU               ! number of grouped response units
+ integer(i4b),intent(out)             :: nHRU               ! number of hydrologic response units
+ integer(i4b),intent(out)             :: err                ! error code
+ character(*),intent(out)             :: message            ! error message
+ ! define general variables
+ character(len=256)                   :: cmessage           ! error message for downwind routine
+ character(LEN=256)                   :: infile             ! input filename
+ 
+ ! define variables for NetCDF file operation
+ integer(i4b)                         :: mode               ! netCDF file open mode
+ integer(i4b)                         :: ncid               ! integer variables for NetCDF IDs
+ integer(i4b)                         :: varid              ! variable id from netcdf file
+ integer(i4b)                         :: hruDimID           ! integer variables for NetCDF IDs
+ integer(i4b)                         :: gruDimID           ! integer variables for NetCDF IDs
+ integer(i4b),allocatable             :: hru_id(:)          ! unique id of hru over entire domain
+ integer(i4b),allocatable             :: gru_id(:)          ! unique ids of GRUs
```

Comment:
I'm wondering if we should be strict on the use of camel back. Underscores can have their place. I've added a comment to the issue #130 to consider updating coding conventions.